include/deal.II/numerics/matrix_tools.h

```diff
@@ -232,23 +234,23 @@ namespace MatrixCreator
    *
    * See the general documentation of this namespace for more information.
    */
-  template <int dim, typename number, int spacedim>
+  template <int dim, int spacedim, typename number>
   void create_mass_matrix (const Mapping<dim, spacedim>       &mapping,
                            const DoFHandler<dim,spacedim>    &dof,
                            const Quadrature<dim>    &q,
                            SparseMatrix<number>     &matrix,
-                           const Function<spacedim> *const a = 0,
+                           const Function<spacedim,number> *const a = 0,
```

Comment:
Yes, it's these kinds of inconsistencies that made it tricky :-( I agree that few people likely used it.

The comparison with FEValues is not entirely accurate. There, we made sure that the output argument had a type that is consistent with all inputs. In some sense, we just made sure that the output has a type that can accommodate the types of all inputs. Here, however, we often have multiple inputs (including, for example, the type of shape functions, coefficients, etc). In principle, we should probably templatize on *all* input types, and enforce the then correct output type. You are correct that that seems like overkill, so let's not worry about it.